src/main/java/chad/TaskList.java

```diff
@@ -0,0 +1,124 @@
+     * @throws ChadException Thrown if description is invalid
+     * @throws IOException Thrown when file cannot be opened in IOException
+     */
+    public static void addTodoTask(ArrayList<Task> tasks, String userInput) throws ChadException, IOException {
+        String outputText = "Got it. I've added this task:\n";
+        String taskDescription = userInput.replaceFirst("todo", "").strip();
+
+        if (taskDescription.isEmpty()) {
+            throw new ChadException("The description of a todo cannot be empty.");
+        }
+        Todo newTask = new Todo(taskDescription);
+        tasks.add(newTask);
+
+        String strIsDone = newTask.getDone() ? "1" : "0";
+        Storage.writeToFile("T | " + strIsDone + " | " + newTask.getDescription());
+
+        outputText += " " + newTask + "\n";
+        outputText += "Now you have " + tasks.size() + " tasks in the list.";
+        Utility.printText(outputText);
+
+    }
+
+    /**
+     * Add deadline task into task list
+     * @param tasks tasks arraylist of tasks
+     * @param userInput deadline task added by user
+     * @throws ChadException Thrown if description or date time is invalid
+     * @throws IOException Thrown when file cannot be opened in IOException
+     */
+    public static void addDeadlineTask(ArrayList<Task> tasks, String userInput) throws ChadException, IOException {
+        String outputText = "Got it. I've added this task:\n";
+
+        String[] temp = userInput.replaceFirst("deadline", "").strip().split("/by");
+        String taskDescription = temp[0].strip();
+        String dateTime = temp[1].strip();
+
+        if (taskDescription.isEmpty()) {
+            throw new ChadException("The description of a deadline cannot be empty.");
+        }
+
+        if (dateTime.isEmpty()) {
+            throw new ChadException("The date of a deadline cannot be empty.");
+        }
+
+        Deadline newTask = new Deadline(taskDescription, dateTime);
+        tasks.add(newTask);
+
+        String strIsDone = newTask.getDone() ? "1" : "0";
+        Storage.writeToFile("D | " + strIsDone + " | " + newTask.getDescription()
+                + " | " + newTask.getDateTime());
+
+        outputText += " " + newTask + "\n";
+        outputText += "Now you have " + tasks.size() + " tasks in the list.";
+        Utility.printText(outputText);
+    }
+
+    /**
+     * Add event task into task list
+     * @param tasks tasks arraylist of tasks
+     * @param userInput event task added by user
+     * @throws ChadException Thrown if description or date time is invalid
+     * @throws IOException Thrown when file cannot be opened in IOException
+     */
+    public static void addEventTask(ArrayList<Task> tasks, String userInput) throws ChadException, IOException {
+        String outputText = "Got it. I've added this task:\n";
+        String[] temp = userInput.replaceFirst("event", "").strip().split("/at");
+        String taskDescription = temp[0].strip();

```

Comment:
Maybe you could consider combining line 101 and line 102 into one ? Since String[] temp is not used elsewhere 🤔